### DISCUSSION TITLE
How to cope with auto renewal function and firewall restrictions

### DISCUSSION THREAD

**Discussion by stefankip** [rcl-ssl/docs#3] [Q&A] [answered]
The staging environments (deployment slots) of our Azure web apps are protected by the firewall based on IP address.
To order the SSL certificate I have configured an 'allow all' rule, so Let's Encrypt can reach the acme challenge.
But I also have the auto renewal Azure Function in place, this will hit the firewall again.

How should I cope with this issue? I can't add any rules to the firewall based on the requested path for example.

**Comment by rcladmin** [comment, marked as answer]
Can you create a certificate now? If you can create a certificate through the firewall both the portal and autorenew function will work. You can also consider using the DNS challenge.
